rtloader/demo/CMakeLists.txt

```diff
@@ -1,11 +1,40 @@
-## Use compiler debugging flags per default
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_DEBUG}")
-
-add_executable(demo main.c)
-include_directories(${CMAKE_SOURCE_DIR}/include)
-if(NOT WIN32)
-    target_link_libraries(demo LINK_PUBLIC datadog-agent-rtloader dl)
-else()
-    set_target_properties(demo PROPERTIES LINK_FLAGS -static)
-    target_link_libraries(demo LINK_PUBLIC datadog-agent-rtloader)
-endif()
+cmake_minimum_required(VERSION 3.19)
+
+project("RTLoader Demo")
+
+option(USE_AGENT_RTLOADER_LIB "Use RTLoader implementation from the agent build" OFF)
+
+## Use compiler debugging flags per default
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_DEBUG}")
+
+add_executable(demo main.c)
+
+## Add our paths for includes. They may be different on the directory we are
+## starting our build from
+if (EXISTS "${CMAKE_SOURCE_DIR}/include")

```

Comment:
If you're trying to build the demo from this repo, the include files are not in `<demo_dir>/include` but in the parent (`<demo_dir>/../include` and common). Given that I didn't quite know how the old values (`<demo_dir>/include`) were used and because there's no downsides to including an `if`/`else` I left in the fallback. If you think the original code was bad, I'll just remove the L14 `true` branch.

Reply:
I missed the fact this was the `CMakeLists.txt` for the demo, not the main project. 🤦 

Answering your question, we probably don't need the original code, just your fix. There's nothing wrong with cleaning up what was wrong :)

Probably what happened is we shuffled/de-duped some code and never really verified this was still building. I'm all 👍 for removing the old broken stuff.